raiden/network/pathfinding.py

```diff
@@ -571,6 +574,16 @@ def query_paths(
     offered_fee = pfs_config.info.price
     scrap_existing_iou = False
 
+    current_info = get_pfs_info(pfs_config.info.url)
+    while current_info.confirmed_block_number < pfs_wait_for_block:
+        log.info(
+            "Waiting for PFS to reach target confirmed block number",
+            pfs_wait_for_block=pfs_wait_for_block,
+            pfs_confirmed_block_number=current_info.confirmed_block_number,
+        )
+        gevent.sleep(0.5)
```

Comment:
I think this sleep might be too short to re-query the PFS for the block number

Reply:
Yeah, I thought about it too. The problem is that if the difference is more than one block, then we have to sleep for more than 7seconds, and if it is one block, we have to have this sort timeout (It is just a normal race, the block exists the PFS just didnt see it yet, so a shorter timeout is better to reduce latency).

Since this is not the hot-path I thought it was okay to have a few additional sleeps for the sake of simpler code.